### PR TITLE
ci: setup for 4.x branch

### PR DESCRIPTION
**Pull request by burkedavison**
Plan:
* Merge this.
* Create 4.x branch.
* Open breaking change PR to update to Spring Cloud 2023 and Spring Boot 3.2

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -59,3 +59,41 @@ updates:
...
+    # Spring Cloud dependencies
+    - dependency-name: "org.springframework.cloud:spring-cloud-dependencies"
+      versions: ["2023.x"]
+    - dependency-name: "org.springframework.cloud:spring-cloud-config"
+      versions: [">=4.1.0"]
+    - dependency-name: "org.springframework.cloud:spring-cloud-function"
+      versions: [">=4.1.0"]
+    # spring-cloud-dependencies-parent should be eventually removed per #1294
+    - dependency-name: "org.springframework.cloud:spring-cloud-dependencies-parent"
+      versions: [">=4.1.0"]
```
Comment (zhumin8) [inline .github/dependabot.yml:96]: nit: I see #1294 closed, is this needed for 4.x?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -59,3 +59,41 @@ updates:
...
+  labels:
+    - "4.x dependencies"
+  ignore:
+    # Ignore major version updates - these should be made manually
+    - dependency-name: "*"
+      update-types: [ "version-update:semver-major" ]
+    # Ignore formatter dependency versions used in codegen modules, since they are fixed for Java 8 support
+    - dependency-name: "com.coveo:fmt-maven-plugin"
+    - dependency-name: "com.google.googlejavaformat:google-java-format"
+    # Ignore dependencies that will be manually upgraded with Spring Boot 3.0 support:
```
Comment (zhumin8) [inline .github/dependabot.yml:81]:
```suggestion
    # Ignore dependencies corresponding to Spring Boot 3.2+, Spring Cloud 2023.x:
```

File: .github/workflows/sonar.yaml [hunk 3]
```diff
@@ -3,7 +3,6 @@ on:
   push:
     branches:
       - main
-      - 3.x
```
Comment (zhumin8) [inline .github/workflows/sonar.yaml:6]: Dropping a note that this will effectively close #2416

**Review by zhumin8 — COMMENTED** [review]
Nothing blocking, dropping a few minor notes.

File: .github/dependabot.yml (hunk 1 above)
Comment (burkedavison) [reply .github/dependabot.yml]: Removed from 4.x configuration

**Review by meltsufin — APPROVED** [review]
You mean, create "4.x branch", not tag, right?

**Comment by burkedavison** [comment]
> You mean, create "4.x branch", not tag, right?

Yes, that's correct. I've fixed the wording.

**Pull request merged**